lib/log/RolarProgressLog.ts

```diff
@@ -113,8 +113,7 @@ export class RolarProgressLog implements ProgressLog {
         const postingLogs = this.localLogs;
         this.localLogs = [];
 
-        if (postingLogs && postingLogs.length > 0) {
-
+        if (isClosed === true || (!!postingLogs && postingLogs.length > 0)) {
```

Comment:
Is including `isClosed === true` actually what we want? It seems that will result in the logs only being posted once when it is closed and the contents will only include the logs between the last flush and the close, since `flush` calls `postLogs` and  the `localLogs` instance variable is cleared every time `postLogs` is called. Plus, the progress log wouldn't be much of a progress log since it would only have content after the goal completed.

Perhaps just not awaiting the promise returned by flush in certain instances is what we want.

Reply:
This is an or between closed and the log entire. Meaning every time there are logs, we’ll write them and we also write closed when there aren’t log entries and we close the log via the close method. 

Not awaiting the promise will drop logs which we don’t want.

Reply:
Sorry, I was reading it as an "and". That makes all the other changes more clear. My mistake.